### PR TITLE
feat(base-cluster): alert rules

### PR DESCRIPTION
**Pull request by jpkraemer-mg**
Add alerting rules for Node Memory / CPU usage

Add alerting rules for HikariCP (max|used) connections

### CODE REVIEW

File: charts/base-cluster/README.md.gotmpl [hunk 1]
```diff
@@ -149,3 +149,7 @@ Please check the following (urgent) upgrade notes before upgrading:
 [k8s upgrade notes v1.27](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#urgent-upgrade-notes)
 
 [redis 7.0 release notes](https://raw.githubusercontent.com/redis/redis/7.0/00-RELEASENOTES) before upgrading.
+
+### To 39.0.1
+
+From 39.0.1 onwards, Prometheus will send alerts for levels where human interference might be necessary in regards to Node CPU / Memory usage as well as HikariCP.
```
Comment (Dominic-Beer) [inline charts/base-cluster/README.md.gotmpl:155]: That this is onwards this version is already stated by the version mentioned above

File: charts/base-cluster/templates/monitoring/rules/node.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            severity: critical
+            frequency: daily
+          annotations:
+            description: The node {{ "{{ $labels.node }}" }} currently uses {{ "{{ $value }}" }}% of their memory, which is above the threshold of 90%
+            summary: High memory usage on {{ "{{ $labels.node }}" }} in namespace {{ "{{ $labels.namespace }}" }}
+    - name: node-cpu
+      rules:
+        - alert: NodeCPUWarning
+          expr: ((1 - sum without (mode, cpu) (rate(node_cpu_seconds_total{job="node-exporter", mode=~"idle|iowait|steal"}[5m])) / ignoring(cpu) group_left count without (cpu, mode) (node_cpu_seconds_total{job="node-exporter", mode="idle"}))) * 100 > 75
+          for: 5m
```
Comment (Dominic-Beer) [inline charts/base-cluster/templates/monitoring/rules/node.yaml:34]: Set to 15m

**Review by Dominic-Beer — CHANGES_REQUESTED** [review]

File: charts/base-cluster/README.md.gotmpl (hunk 1 above)
Comment (jpkraemer-mg) [reply charts/base-cluster/README.md.gotmpl]: Changed

File: charts/base-cluster/templates/monitoring/rules/node.yaml (hunk 2 above)
Comment (jpkraemer-mg) [reply charts/base-cluster/templates/monitoring/rules/node.yaml]: Changed

File: charts/base-cluster/templates/monitoring/rules/hikaricp.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          for: 15m
+          labels:
+            severity: warning
+
+        - alert: HikariCPAlmostFull
+          annotations:
+            description: Hikari Connection Pool for {{ "{{ $labels.node }}" }} in namespace {{ "{{ $labels.namespace }}" }} is above 90% longer than 5 minutes.
+            summary: HikariCP filled halfway for 5 minutes in {{ "{{ $labels.namespace }}" }} / {{ "{{ $labels.node }}" }}
+          expr: max by (job, namespace) (ceil((hikaricp_connections{job=~".*-backend", namespace!="monitoring"}/ hikaricp_connections_max{job=~".*-backend", namespace!="monitoring"}) * 100)) > 90
+          for: 5m
```
[the remark this answers is not in the document]
Comment (jpkraemer-mg) [reply charts/base-cluster/templates/monitoring/rules/hikaricp.yaml]: Time decreased to 30s

**Review by Dominic-Beer — APPROVED** [review]

**Review by Dominic-Beer — APPROVED** [review]

**Review by Dominic-Beer — APPROVED** [review]

**Review by C-schwarz-4ap — APPROVED** [review]

**Review by C-schwarz-4ap — APPROVED** [review]

**Review by Dominic-Beer — APPROVED** [review]

**Review by Dominic-Beer — APPROVED** [review]

**Review by C-schwarz-4ap — APPROVED** [review]

**Review by Dominic-Beer — APPROVED** [review]

**Review by C-schwarz-4ap — DISMISSED** [review]

**Review by Dominic-Beer — APPROVED** [review]

**Pull request merged**